src/duration.rs

```diff
@@ -443,14 +443,42 @@ pub struct Generic<T> {
     scaling_factor: Fraction,
 }
 
+impl<T: TimeInt> Generic<T> {
+    /// Try to create a new, equivalent `Generic` with the given _scaling factor_
+    fn try_into_scaling_factor(self, scaling_factor: Fraction) -> Result<Self, ConversionError> {
+        let new_int = TimeInt::checked_mul_fraction(
+            &self.integer,
+            &self
+                .scaling_factor
+                .checked_div(&scaling_factor)
+                .ok_or(ConversionError::Overflow)?,
+        )
+        .ok_or(ConversionError::Overflow)?;
+
+        Ok(Self::new(new_int, scaling_factor))
+    }
+}
+
 impl<T: TimeInt> PartialOrd<Generic<T>> for Generic<T> {
     /// See [Comparisons](trait.Duration.html#comparisons)
     fn partial_cmp(&self, rhs: &Generic<T>) -> Option<core::cmp::Ordering> {
-        Some(
-            self.integer
-                .checked_mul_fraction(&self.scaling_factor)?
-                .cmp(&rhs.integer.checked_mul_fraction(&rhs.scaling_factor)?),
-        )
+        if self.scaling_factor == rhs.scaling_factor {
+            Some(self.integer.cmp(&rhs.integer))
+        } else if self.scaling_factor < rhs.scaling_factor {
+            // convert to the smaller scaling factor (rhs -> self)
+            // if conversion fails, we know self is less than rhs
```

Comment:
It feels like this will be true most of the time, but I'm not entirely convinced that there aren't edge cases where `rhs.scaling_factor / self.scaling_factor < self.integer / rhs.integer` (so `self > rhs`), and `rhs` is close enough to the maximum bound on `T` that it will overflow. In particular, `TimeInt::checked_mul_fraction` is being called with `rhs.integer` and a fraction that is greater than 1, and is implemented by first multiplying by the numerator (which could overflow) and then dividing by the denominator (which if not for the overflow could bring the result back in range and below `self.integer`). It would be great to have tests that exercise these near-bounds edge cases.

Reply:
I have stopped contributing here and created [`fugit`](https://crates.io/crates/fugit) instead.